src/Sulu/Bundle/SecurityBundle/Entity/Role.php

```diff
@@ -93,14 +98,9 @@ public function getId()
         return $this->id;
     }
 
-    public function getRole()
-    {
-        return 'ROLE_SULU_' . strtoupper($this->name);
-    }
-
     public function getIdentifier()
     {
-        return 'ROLE_SULU_' . strtoupper($this->getName());
+        return 'ROLE_SULU_' . strtoupper($this->getKey());
```

Comment:
I would make the key nullable and fallback to the name makes the migration simplier and doesn't crash projects where developer did just `doctrine:schema:update` or where developer did build custom logic with ROLE_USER_{NAME}.

```suggestion
$key = $this->getKey();

if (!$key) {
    $key = $this->getName();
}

return 'ROLE_SULU_' . strtoupper($key);
```

Reply:
Also think its important to have the key nullable know as I think there are synchronisations with e.g Active Directories which will only update the role `NAME` and not the key. So I think its important to have the key nullable and fallback to the name in the getIdentifier method.